packages/test-utils/index.ts

```diff
@@ -33,8 +34,15 @@ export function startLanguageServer(serverModule: string, cwd?: string | URL) {
 	connection.onClose(e => console.log(e));
 	connection.onUnhandledNotification(e => console.log(e));
 	connection.onError(e => console.log(e));
+	connection.onNotification(_.LogMessageNotification.type, e => {
+		if (e.type === _.MessageType.Error || e.type === _.MessageType.Warning) {
+			console.error(e.message);
+		} else {
+			console.log(e.message);
```

Comment:
`console.log` outputs to `stdout` so this will likely interfere with LSP communication and cause crashes. I think all notification types need to go through `console.error`.

Reply:
This was once true. That’s why nowadays `vscode-languageserver` intercepts console methods and redirects them to the server connection logger, which sends them as `window/logMessage` notifications (https://github.com/microsoft/vscode-languageserver-node/pull/1301). So it’s actually safe to use the console in Node.js language servers now.

This code however, is client code. It receives these `window/logMessage` notifications and logs the message to the console using the proper output stream.

Reply:
OK. Thanks for explaining.

(I can't close this issue myself)